crowbar_framework/app/controllers/backup_controller.rb

```diff
@@ -0,0 +1,162 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class BackupController < ApplicationController
+  skip_before_filter :enforce_installer
+
+  def index
+    @backups = Backup.all_images
+  end
+
+  #
+  # Backup
+  #
+  # Provides the restful api call for
+  # /utils/backup   POST   Trigger a backup
+  def backup
+    bu = Backup.create(params[:filename])
+    respond_to do |format|
+      if !bu.nil? && bu.save
+        format.json { head :ok }
+        format.html { redirect_to backup_path }
+      else
+        msg = I18n.t(".invalid_filename", scope: "backup.index")
+        format.json { render json: msg }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+      end
+    end
+  end
+
+  #
+  # Restore
+  #
+  # Provides the restful api call for
+  # /utils/backup/restore   POST   Trigger a restore
+  def restore
+    bu = Backup.new(params[:name], params[:created_at])
+
+    respond_to do |format|
+      if bu.save
+        ret = bu.restore
+        byebug
+        if ret[:status] == :ok
+          format.json { head :ok }
+          format.html { redirect_to backup_path }
+        else
+          format.html do
+            flash[:alert] = ret[:msg]
+            redirect_to backup_path
+          end
+          format.json { render json: ret[:msg] }
+        end
+      else
+        msg = I18n.t(".invalid_backup", scope: "backup.index")
+        format.html do
+          flash[:alert] = msg
+          redirect_to backup_path
+        end
+        format.json { render json: msg }
+      end
+    end
+  end
+
+  #
+  # Backups
+  #
+  # Provides the restful api call for
+  # /utils/backups 	GET 	Returns a json list of available backups
+  def backups
+    # read the contents of /var/lib/crowbar/backup or rails.root("storage")
+    respond_to do |format|
+      format.html { redirect_to backup_path }
+      format.json { render json: Backup.all_images.to_json }
+    end
+  end
+
+  #
+  # Download
+  #
+  # Provides the restful api call for
+  # /utils/backup/download/:name/:created_at 	GET 	Download a backup
+  def download
+    bu = Backup.new(params[:name], params[:created_at])
```

Comment:
Beside that I would prefer a similar interface like active model, e.g. the parameter to `initialize` is an hash.